client/share_trader/js/components/assets.js

```diff
@@ -1,211 +1,57 @@
-        handleAssetClick(asset);
-    },
-
-    handleTransferClick() {
-        const { asset, activeAccount } = this.props;
-        const { selectedAccount } = this.state;
-
-        const idToTransfer = {
-            txid: asset.id,
-            cid: 0
-        };
-
-        const payloadToPost = {
-            'source': activeAccount,
-            'to': selectedAccount
-        };
-
-        AssetActions.transferAsset({
-            idToTransfer,
-            payloadToPost
-        });
-
-        this.setState({ transfered: true });
-    },
-
-    render() {
-        const { asset, active, activeAccount, accountList, assetClass } = this.props;
-        const { selectedAccount, transfered } = this.state;
-
-        const { data: { payload: { content } } = {} } = asset.transaction;
-        const validGlyph = asset.hasOwnProperty('assignee') ? <Glyphicon glyph="cog" />
-                                                            : <Glyphicon glyph="ok" />;
-
-        let actionsPanel = null;
-        if (active && activeAccount && accountList && !transfered) {
-            actionsPanel = (
-                <AssetActionPanel
-                    accountList={accountList}
-                    activeAccount={activeAccount}
-                    handleAccountClick={this.setSelectedAccount}
-                    handleTransferClick={this.handleTransferClick}
-                    selectedAccount={selectedAccount} />
-            );
-        }
+export default function Assets(props) {
+    const {
+        activeAccount,
+        accountList,
+        assetList,
+        activeAsset,
+        assetClasses,
+        handleAssetClick
+    } = props;
 
+    if (assetList && assetList.length) {
         return (
-            <Row onClick={this.handleAssetClick}>
-                <div
-                    className={classNames('asset-container', assetClass, {
-                        transfered,
-                        active: active && !transfered
-                    })}>
-                    <div className="asset-container-id">
-                        {asset.id}
-                    </div>
-                    <div className="asset-container-detail">
-                        {content ? `Row: ${content.y + 1}, Col: ${content.x + 1}` : '-'}
-                    </div>
-                    <div className="asset-container-timestamp">
-                        {new Date(parseInt(asset.transaction.timestamp, 10) * 1000).toGMTString() + '   '}
-                        {validGlyph}
-                    </div>
-                    {actionsPanel}
-                </div>
-            </Row>
+            <div>
+                {assetList.sort((a, b) => a.transaction.timestamp - b.transaction.timestamp)
+                    .map((asset) => {
+                        const active = (activeAsset) ? activeAsset.id === asset.id : false;
+                        const assetClass = assetClasses[asset.transaction.conditions[0]
```

Comment:
Will this always be there or is this from IO? Maybe it might be good if the data is actually available?